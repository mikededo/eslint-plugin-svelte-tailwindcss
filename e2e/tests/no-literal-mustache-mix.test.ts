import { describe, expect, it } from 'vitest';

import { isValidTest, lintResult } from '../helpers';
import { lintFile } from '../setup';

describe('no-literal-mustache-mix rule', () => {
  it('valid', async () => {
    await isValidTest('valid/no-literal-mustache-mix.svelte');
  });

  it('invalid', async () => {
    const res = await lintFile('invalid/no-literal-mustache-mix.svelte');
    expect(lintResult(res)).toMatchSnapshot();
  });
});

