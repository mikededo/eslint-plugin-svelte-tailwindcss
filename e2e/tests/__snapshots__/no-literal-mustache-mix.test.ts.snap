// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-literal-mustache-mix rule > invalid 1`] = `
{
  "errorCount": 1,
  "fixableErrorCount": 1,
  "fixableWarningCount": 0,
  "messages": [
    {
      "column": 6,
      "endColumn": 48,
      "endLine": 1,
      "fix": {
        "range": [
          12,
          46,
        ],
        "text": "bg-blue-500 py-4 {twMerge('px-8')}",
      },
      "line": 1,
      "message": "Do not mix literal expressions with mustache expressions",
      "messageId": "no-mix",
      "nodeType": "SvelteAttribute",
      "ruleId": "svelte-tailwindcss/no-literal-mustache-mix",
      "severity": 2,
    },
  ],
  "warningCount": 0,
}
`;
